chapter/7/langs-consistency.md

```diff
@@ -1,11 +1,270 @@
+
+Why not push the consistency guarantees in between the IO-level and the application-level? Is there any reason why you as the programmer couldn't program using tools that facilitate these types of monotonic programs? If you're familiar with formal systems -- why not construct a formal system (programming language / library) in which every theorem (program) is formally guarunteed to be monotonic? If it's *impossible* to express a non-monotonic program, the programmer needn't worry about maintaining a direct mapping between their code and their mental model.
+
+Wouldn't it be great if tools like this existed?
+
+### Bloom
+[ Introduce Bloom ]
+
+#### Restriction & Danger
+[Bloom restricts you, it's different, and it's dangerous]
+
+### Lasp
+[ Introduce Lasp ]
+Instead of trying to do it all (and accepting danger), it tries to be embeddable (and truly restrictive.)
+
+### Utilization
+
+Lasp is an Erlang library, and for good reason. Remember the initial discussion and reasoning for models such as Bloom and Lasp: we have a specific type of application that doesn't require tight consistency constraints. The constraints that do exist have been formalized, and we can be quite sure that by using a DSL like Lasp, we'll be safe from interns like Jerry. But Lasp can't do everything. More generally, eventual consistency doesn't solve every problem.
+
+PostgreSQL enforce very specific and restrictive IO-level consistency, and this was too much for our needs. But it's certainly not too much for *all* needs. There certainly are applications (take banking, for example) in which consistency is extremely important. You certainly are not allowed to double spend your money depending on how fast you can travel to a different server, so eventual consistency is not enough! All servers must coordinate.
+
+There's a key principle here, however: distributed programming models that attempt to accomdate everything end up doing nothing well; models that accept compromises and formalize certain properties end up being extremely useful for a subset of domains.
+
+Most programming languages are "general-use". This works for single machine programming. As the world moves toward distributed programming, programmers must adopt models / languages / libraries that are built for their domain. It forces serious thought on the part of the programmer: what *exactly* am I trying to achieve, and what am I willing to sacrifice?
+
+We've known for quite a while that when we're talking about multiple machines, we can't have it all. Our tools must now reflect this mantra. Our sanity and the safety of our programs depends on it.
 
 ## References
```

Comment:
I realize I've gotten to the end and that there are many language-integrated things that are similar to CRDTs haven't been mentioned. LVars and FlowPools are two such examples. I'm sure there are more; Chris's references in his most recent paper would probably be the most up-to-date on what's out there and on what's similar.